### PR TITLE
async send message

### PR DESCRIPTION
**Pull request by nicolasparada**
In the test I was sending 10 messages at once and measuring how much it takes.

Before: around 10ms.
After: around 30µs.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205602086260264

### CODE REVIEW

File: cshared.go [hunk 1]
```diff
@@ -130,30 +131,16 @@ func FLBPluginInputCallback(data *unsafe.Pointer, csize *C.size_t) int {
...
 
 		t := input.FLBTime{Time: msg.Time}
 		b, err := input.NewEncoder().Encode([]any{t, msg.Record})
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "encode: %s\n", err)
-			return input.FLB_ERROR
+			return
 		}
 
 		cdata := C.CBytes(b)
```
Comment (pwhelan) [inline cshared.go:146]: Each time send is executed another buffer will be leaked. Only a single buffer will be received by the calyptia side in each polling interval (1000 nanoseconds). Only the buffer that actually gets received will be freed.

**Review by pwhelan — CHANGES_REQUESTED** [review]
At the very least a test needs to be added to make sure that messages are not being skipped, as I pointed out in my review.

File: cshared.go [hunk 2]
```diff
@@ -162,6 +149,20 @@ func FLBPluginInputCallback(data *unsafe.Pointer, csize *C.size_t) int {
 		*csize = C.size_t(len(b))
 
 		// C.free(unsafe.Pointer(cdata))
+	}
```
[the remark this answers is not in the document]
Comment (nicolasparada) [reply cshared.go:152]: True. It's something that should be fixed in the fluent-bit side I think.
For now this fixes the time-shift issue that Devo is having... Where the request parameters that use timing, are getting behind due to slow sending of the messages to fluent-bit.

File: plugin_test.go [hunk 3]
```diff
@@ -139,14 +139,26 @@ func testPlugin(t *testing.T, pool *dockertest.Pool) {
...
-					Foo            string   `json:"foo"`
-					Message        string   `json:"message"`
-					TmplOut        string   `json:"tmpl_out"`
-					MultilineSplit []string `json:"multiline_split"`
+					SkipMe               bool     `json:"skip_me"`
+					Foo                  string   `json:"foo"`
+					Message              string   `json:"message"`
+					TmplOut              string   `json:"tmpl_out"`
+					MultilineSplit       []string `json:"multiline_split"`
+					TookToSend10Messages string   `json:"took_to_send_10_messages"`
```
[the remark this answers is not in the document]
Comment (nicolasparada) [reply plugin_test.go:147]: I'm actually ignoring these the 10 consecutive messages on this test. Those 10 are dummy to measure how much it takes to send them. Then an 11th message is sent with the time it took to send those 10 dummy ones.

**Pull request closed without merging**